### ISSUE TITLE
Index should check that there are no duplicate read names

### ISSUE THREAD

**Issue by mbhall88** [rmcolq/pandora#132]
My feeling is that if we see two sequences with the same name we should raise an error. A lot of other bioinf tools also exhibit this behaviour. I think it also saves bigger unexplained issues (like this one) arising further down the analysis road.

_Originally posted by @mbhall88 in https://github.com/rmcolq/pandora/issues/126#issuecomment-492161063_

From the discussion in #126 it would save users a lot of heartache further down the analysis pipeline if we catch any entries in the PRG which have the same name and exit with an error saying all names should be unique. This is effectively a proxy for ensuring that the same PRG is not in a PRG file twice.

**Comment by mbhall88** [comment]
@leoisl given you are working on `index` at the moment maybe it makes sense for you to add this in?

**Comment by leoisl** [comment]
Yeah, I will take care of it after #23

**Comment by leoisl** [comment]
Update: I am not taking care of this right after #23 

This has low priority, moving towards finishing learning `pandora index, map and compare` to try to make the most of the fact that Rachel is still here (without her, I am sure it would be way slower for me to understand `index`).
